src/embedCreators/BaseEmbedCreator.ts

```diff
@@ -0,0 +1,30 @@
+﻿import { EventData } from '../models/internal-models';
+import { Lang } from '../services/index.js';
+import { EmbedBuilder } from 'discord.js';
+
+export default class BaseEmbedCreator {
+    protected _embed: EmbedBuilder;
+    protected _fields : IEmbedCreatorField[] = [];
+
+    constructor(languageLocation: string, data: EventData, variables?: {}) {
+        this._embed = Lang.getEmbed(languageLocation, data.langGuild, variables ?? {});
+    }
+    
+    setNameAndDescription (name: string, description: string) {
+        this._embed.setTitle(name);
+        this._embed.setDescription(description);
+    }
+
+    addField(field: IEmbedCreatorField){
```

Comment:
return this, so that the functions can be chained. Making actually a builder-pattern, so instead of making it as creator, rename to builder.